packages/data-context/src/sources/ProjectDataSource.ts

```diff
@@ -335,17 +336,46 @@ export class ProjectDataSource {
     // We respond to all changes to the project's filesystem when
     // files or directories are added and removed that are not explicitly
     // ignored by config
-    this._specWatcher = chokidar.watch('.', {
-      ignoreInitial: true,
-      ignorePermissionErrors: true,
-      cwd: projectRoot,
-      ignored: ['**/node_modules/**', ...excludeSpecPattern, ...additionalIgnorePattern],
+    this._specWatcher = this._makeSpecWatcher({
+      projectRoot,
+      specPattern,
+      excludeSpecPattern,
+      additionalIgnorePattern,
     })
 
     // the 'all' event includes: add, addDir, change, unlink, unlinkDir
     this._specWatcher.on('all', onProjectFileSystemChange)
   }
 
+  _makeSpecWatcher ({ projectRoot, specPattern, excludeSpecPattern, additionalIgnorePattern }: { projectRoot: string, excludeSpecPattern: string[], additionalIgnorePattern: string[], specPattern: string[] }) {
+    return chokidar.watch('.', {
+      ignoreInitial: true,
+      ignorePermissionErrors: true,
+      cwd: projectRoot,
+      ignored: ['**/node_modules/**', ...excludeSpecPattern, ...additionalIgnorePattern, (file: string, stats?: fs.Stats) => {
+        // We need stats arg to make the determination of whether to watch it, because we need to watch directories
+        // chokidar is extremely inconsistent in whether or not it has the stats arg internally
```

Comment:
I also saw some comments around `**/node_modules/**` not always working as expected: https://github.com/paulmillr/chokidar/issues/773

Could do [like this](https://github.com/paulmillr/chokidar/issues/773#issuecomment-536793469)

```js
if (file.includes('node_modules')) {
  return true
}
```

To be extra safe?